In Refrence to Issue #2

astoycos opened the pull request:
Remove -tlsConfig flag

Enable a TLS connection if user supplies an "amqps" connection endpoint

Provide more Error Handling->
        Throw Error if user tries to pass --insecure=true or --tlsPath for a non TLS connection
        Throw Error if connection is TLS but the  user Dosn't pass either --insecure=true nor a TLS Path

TODO: Apply the TLS setup to http and mqtt  publish functions following approval of the general design by @ctron 

Let me know what you think about this design, glad to change it around if you see fit!

astoycos committed:
In refrence to Issue#2

Remove -tlsConfig flag

Enable a TLS connection if user supplies an "amqps" connection endpoint

Provide more Error Handling->
        Throw Error if user tries to pass --insecure=true or --tlsPath for a non TLS connection
        Throw Error if connection is TLS but the  user Dosn't pass either --insecure=true nor a TLS Path

TODO: Apply the TLS setup to http and mqtt  publish functions following approval of the general design

astoycos committed:
Update readme for PR, Remove --tlsConfig

ctron commented on cmd/main.go:
```diff
@@ -72,9 +73,16 @@ func main() {
+			if (args[1][0:5] != "amqps" && insecure) {
 				return errors.New("Cannot have an insecure TLS connection without TLS enabled")
 			}
+			if (args[1][0:5] != "amqps" && tlsPath != "") {
```
I think it is possible to have TLS without a cert. I would expect that in this case the system defaults will get used.

ctron commented on cmd/main.go:
```diff
@@ -72,9 +73,16 @@ func main() {
 				return errors.New("Wrong number of Input arguments expected 3 got "+strconv.Itoa(len(args)))
 			}
-			if (tlsConn == false && insecure == true) {
+			if (args[1][0:5] != "amqps" && insecure) {
```
I don't think we need special support here, as this is already handled by the AMQP library:

https://github.com/vcabbage/amqp/blob/5a75e78e59d314879580d023e789feb1d4d1d22b/client.go#L77-L80

Commit:
Update for Pull Request

Rely on amqp client defaults if the user does not request anly TLS configurations for an "amqps" connection

Update Readme to reflect these changes

Left error checking for if user tries to set TLS flags with a non TLS connection to help make usability a bit clearer and help with any accidentaluser mistakes

ctron commented:
IIRC we fixed this in another PR.
